### PR TITLE
Fix expected field headers format

### LINKED ISSUE #86: 'Reply-To address not added because From address was not properly extracted.'
In some cases the email that is parsed contains
"From:<user@example.com>"
In these cases the extraction of the From: does not work because the regular expression expects a space after :
In my project I fixed that by removing the space after From:  in the regular expression in lines 190 and 205. 

https://github.com/arithmetric/aws-lambda-ses-forwarder/blob/a04bf17228f4ee88721d9c7fd22413045cf0909c/index.js#L190

### PR DESCRIPTION
**Pull request by ccjmne**
Fixes #86.

As described in the [RFC 5322 – Internet Message Format](https://www.rfc-editor.org/rfc/rfc5322.txt), `from:Betsy <betsy@example.com>` should be accepted and understood.

Basically, fields don't have to look like `From: ccjmne@gmail.com`, they could very well be: `from:ccjmne@gmail.com` (note the **non-capitalized** field header and the **absence of a whitespace** immediately following the colon `:`).

I ensured this would work for the relevant fields, and slightly modified the `test/assets/message.txt` file to reproduce the uncommon **"lowercase header, colon, no-space, body"** field pattern.

---

Also, according to that same RFC, the "carriage return/line feed" pair should **always** appear together.
In other words, `\r?\n` should really be: `\r\n` ; but I figure it's safe being too lenient there so I'll leave these unchanged.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -186,8 +186,8 @@ exports.processMessage = function(data) {
   var body = match && match[2] ? match[2] : '';
 
   // Add "Reply-To:" with the "From" address if it doesn't already exists
-  if (!/^Reply-To: /mi.test(header)) {
-    match = header.match(/^From: (.*(?:\r?\n\s+.*)*\r?\n)/m);
+  if (!/^reply-to:\s?/mi.test(header)) {
```
Comment (ccjmne) [inline index.js]:
```suggestion
  if (!/^reply-to:[\t ]?/mi.test(header)) {
```

File: index.js [hunk 2]
```diff
@@ -202,7 +202,7 @@ exports.processMessage = function(data) {
   // so replace the message's "From:" header with the original
   // recipient (which is a verified domain)
   header = header.replace(
-    /^From: (.*(?:\r?\n\s+.*)*)/mg,
+    /^from:\s?(.*(?:\r?\n\s+.*)*)/mgi,
```
Comment (ccjmne) [inline index.js]:
```suggestion
    /^from:[\t ]?(.*(?:\r?\n\s+.*)*)/mgi,
```

File: index.js [hunk 3]
```diff
@@ -186,8 +186,8 @@ exports.processMessage = function(data) {
   var body = match && match[2] ? match[2] : '';
 
   // Add "Reply-To:" with the "From" address if it doesn't already exists
-  if (!/^Reply-To: /mi.test(header)) {
-    match = header.match(/^From: (.*(?:\r?\n\s+.*)*\r?\n)/m);
+  if (!/^reply-to:\s?/mi.test(header)) {
+    match = header.match(/^from:\s?(.*(?:\r?\n\s+.*)*\r?\n)/mi);
```
Comment (ccjmne) [inline index.js]:
```suggestion
    match = header.match(/^from:[\t ]?(.*(?:\r?\n\s+.*)*\r?\n)/mi);
```

File: index.js (hunk 1 above)
Comment (ccjmne) [inline index.js]:
```suggestion
  if (!/^reply-to:[\t ]?/mi.test(header)) {
```

File: index.js [hunk 4]
```diff
@@ -218,31 +218,31 @@ exports.processMessage = function(data) {
   // Add a prefix to the Subject
   if (data.config.subjectPrefix) {
     header = header.replace(
-      /^Subject: (.*)/mg,
+      /^subject:\s?(.*)/mgi,
```
Comment (ccjmne) [inline index.js]:
```suggestion
      /^subject:[\t ]?(.*)/mgi,
```

File: index.js [hunk 5]
```diff
@@ -218,31 +218,31 @@ exports.processMessage = function(data) {
...
       function(match, subject) {
         return 'Subject: ' + data.config.subjectPrefix + subject;
       });
   }
 
   // Replace original 'To' header with a manually defined one
   if (data.config.toEmail) {
-    header = header.replace(/^To: (.*)/mg, () => 'To: ' + data.config.toEmail);
+    header = header.replace(/^to:\s?(.*)/mgi, () => 'To: ' + data.config.toEmail);
```
Comment (ccjmne) [inline index.js]:
```suggestion
    header = header.replace(/^to:[\t ]?(.*)/mgi, () => 'To: ' + data.config.toEmail);
```

File: index.js [hunk 6]
```diff
@@ -218,31 +218,31 @@ exports.processMessage = function(data) {
...
   }
 
   // Remove the Return-Path header.
-  header = header.replace(/^Return-Path: (.*)\r?\n/mg, '');
+  header = header.replace(/^return-path:\s?(.*)\r?\n/mgi, '');
```
Comment (ccjmne) [inline index.js]:
```suggestion
  header = header.replace(/^return-path:[\t ]?(.*)\r?\n/mgi, '');
```

File: index.js [hunk 7]
```diff
@@ -218,31 +218,31 @@ exports.processMessage = function(data) {
...
 
   // Remove Sender header.
-  header = header.replace(/^Sender: (.*)\r?\n/mg, '');
+  header = header.replace(/^sender:\s?(.*)\r?\n/mgi, '');
```
Comment (ccjmne) [inline index.js]:
```suggestion
  header = header.replace(/^sender:[\t ]?(.*)\r?\n/mgi, '');
```

File: index.js [hunk 8]
```diff
@@ -218,31 +218,31 @@ exports.processMessage = function(data) {
...
 
   // Remove Message-ID header.
-  header = header.replace(/^Message-ID: (.*)\r?\n/mig, '');
+  header = header.replace(/^message-id:\s?(.*)\r?\n/mgi, '');
```
Comment (ccjmne) [inline index.js]:
```suggestion
  header = header.replace(/^message-id:[\t ]?(.*)\r?\n/mgi, '');
```

File: index.js [hunk 9]
```diff
@@ -218,31 +218,31 @@ exports.processMessage = function(data) {
...
 
   // Remove all DKIM-Signature headers to prevent triggering an
   // "InvalidParameterValue: Duplicate header 'DKIM-Signature'" error.
   // These signatures will likely be invalid anyways, since the From
   // header was modified.
-  header = header.replace(/^DKIM-Signature: .*\r?\n(\s+.*\r?\n)*/mg, '');
+  header = header.replace(/^dkim-signature:\s?.*\r?\n(\s+.*\r?\n)*/mgi, '');
```
Comment (ccjmne) [inline index.js]:
```suggestion
  header = header.replace(/^dkim-signature:[\t ]?.*\r?\n(\s+.*\r?\n)*/mgi, '');
```

**Comment by alejandroclose** [comment]
Great fix @ccjmne! 
@arithmetric  Thank you for the original code! 
The forwarder kept giving me the error: ` 'sendRawEmail() returned error.', InvalidParameterValue: Empty header names are illegal. ` until I applied these changes. 
#KeepRowing

**Comment by ccjmne** [comment]
Awesome!
I'm glad it worked for you 😄

**Comment by arithmetric** [comment]
Thanks for this contribution @ccjmne!
